File: Universe.cpp
#include "Universe.h"

Universe::Universe(){
    amountOfColumns = 0;
    amountOfRows = 0;
}

Universe::Universe(int amountOfRows,int amountOfColumns)
{
	this->amountOfRows=amountOfRows;
	this->amountOfColumns=amountOfColumns;
}

int Universe::getAmountOfRows()
{
	return amountOfRows;
}

int Universe::getAmountOfColumns()
{
	return amountOfColumns;
}

void Universe::setAmountOfRows(int amountOfRows)
{
	this->amountOfRows=amountOfRows;
}

void Universe::setAmountOfColumns(int amountOfColumns)
{
	this->amountOfColumns=amountOfColumns;
}

File: java/Universe.java
public abstract class Universe{
	private int amountOfRows;
	private int amountOfColumns;

	public Universe(){
		amountOfRows = 0;
		amountOfColumns = 0;
	}
	
	public Universe(int rows,int columns){
		amountOfRows = rows;
		amountOfColumns = columns;
	}

	public int getAmountOfRows(){
		return amountOfRows;
	}
	
	public int getAmountOfColumns(){
		return amountOfColumns;
	}

	void setAmountOfRows(int rows){
		amountOfRows = rows;
	}
	
	void setAmountOfColumns(int columns){
		amountOfColumns = columns;
	}

	/**
	*Kill a creature from CreatureList
	*@param C a pointer to creature
	**/
//	abstract void killCreature(Creature C);
	abstract void killCreature(int C);

	/**
	*Add a creature to CreatureList
	*@param C a pointer to creature
	**/
//	abstract void addCreature(Creature C);
	abstract void addCreature(int C);

	/**
	*Add "amount" amount of random creatures to CreatureList
	*@param amount an integer
	**/
	abstract void addRandomCreature(int amount);

	/**
	*return 1 if Universe is empty, 0 otherwise
	**/
	abstract int isWorldEmpty();

	/**
	*check for a collision or out of bounds creatures to be killed
	**/
	abstract void checkForCollisions();


	/**
	*Move all creature from CreatureList once
	**/
	abstract void moveAllCreaturesOnce();

	/**
	*Print the world and all its creatures
	**/
	abstract void printFile(String str);
	
	abstract void print();
}